DTML.EduBot/Dialogs/RootDialog.cs

```diff
@@ -99,111 +106,29 @@ public Task StartAsync(IDialogContext context)
+            var messageActivity = await result;
+            await context.Forward(_chitChatDialog, this.AfterChitChatComplete, messageActivity, CancellationToken.None);
         }
 
         private async Task UserNameReceivedInNativeLanguageAsync(IDialogContext context, IAwaitable<IMessageActivity> result)
         {
             string nativeLanguageIsoCode = MessageTranslator.DEFAULT_LANGUAGE;
-            using (var scope = WebApiApplication.FindContainer().BeginLifetimeScope())
+            using(var scope = WebApiApplication.FindContainer().BeginLifetimeScope())
             {
                 UserData userData = scope.Resolve<IUserDataRepository>().GetUserData(context.Activity.From.Id);
                 nativeLanguageIsoCode = userData.NativeLanguageIsoCode;
             }
 
-            string translatedWhatToDo =
-                await MessageTranslator.TranslateTextAsync(
-                    $"Hello Dear {result.GetAwaiter().GetResult().Text},\n What would you like to do.", nativeLanguageIsoCode);
-
-            string translatedNotUnderstandSelection =
-                await MessageTranslator.TranslateTextAsync(
-                    "I am sorry but I didn't understand that. I need you to select one of the options below", nativeLanguageIsoCode);
-
-            PromptDialog.Choice(
-                context,
-                this.AfterDialogChoiceSelectedInNativeLanguageAsync,
-                await MessageTranslator.TranslatedChoices(DialogChoices, nativeLanguageIsoCode),
-                translatedWhatToDo,
-                translatedNotUnderstandSelection,
-                attempts: Shared.MaxAttempt);
-        }
-
-        private async Task AfterDialogChoiceSelectedAsync(IDialogContext context, IAwaitable<string> result)
-        {
-            try
-            {
-                var selection = await result;
-
-                using (var scope = WebApiApplication.FindContainer().BeginLifetimeScope())
-                {
-                    switch (selection)
-                    {
-                        case Shared.ChatWithBot:
-                            await context.PostAsync("Great! Say Hello, and see what will I respond!");
-                            context.Call(scope.Resolve<ChitChatDialog>(), this.AfterDialogEnded);
-                            break;
-
-                        case Shared.StartTheLessonPlan:
-                            context.Call(scope.Resolve<LevelDialog>(), this.AfterDialogEnded);
-                            break;
-                    }
-                }
-            }
-            catch (TooManyAttemptsException)
-            {
-                await this.StartAsync(context);
-            }
+            IMessageActivity userText = await result;
+            string userTextInEnglish = await MessageTranslator.TranslateTextAsync(userText.Text);
+            await context.Forward(_chitChatDialog, this.AfterChitChatComplete, new Activity { Text = userTextInEnglish }, CancellationToken.None);
```

Comment:
It is not a good idea to instantiate a new instance of Activity explicitly just to set the Text in English. Note IMessageActivity contains:

string Text { get; set; }

that allows you to set the text.

Plus, Activity is the basic communication protocol between bot and channel, so another reason not a good idea to instantiate just to forward to a different dialog.